### ISSUE TITLE
Background AppleScript causing BTT to freeze when triggered from a Floating Web View

### ISSUE THREAD

**Issue by CalderH** [folivoraAI/BetterTouchTool#2714]
#### Description of bug/feature request/question:

I was experimenting with using a Floating Web View to display the volume in a less intrusive way than the rectangle that usually appears in the middle of the screen. I followed the instructions [here](http://docs.bettertouchtool.net/docs/floating_html_menu.html) to make a named trigger run an AppleScript that returns the volume, and then to use that value in the HTML file. In the instructions, a screenshot shows the “Run Apple Script” action selected in the Other tab, but I found that, when I use that option, the callback function only ever receives null. When I use “Run Apple Script in background,” it does receive the volume, and the Floating Web View works as I intended. However, with this setup, when I activate the Floating Web View a few times, something causes BTT to freeze, and I have to force quit it.

#### Affected input device (e.g. MacBook Trackpad, Magic Mouse/Trackpad, Touch Bar, etc.):

Floating Web View (triggered by Touch Bar gesture)

#### Device information:

Type of Mac: MacBook Pro (2016, Touch Bar)
macOS version: 10.13
BetterTouchTool version: 2.500

#### Additional information (e.g. StackTraces, related issues, screenshots, workarounds, etc.):

(none)

**Comment by fifafu** [comment]
hey,

could you please post the exact Apple Script you are using?

Possibly try checking for updates, the 2.503 version may contain some related fixes (but I haven't experienced freezing so it may not help in that case)

**Comment by CalderH** [comment]
Sure. I have `return (output volume of (get volume settings))`.
Also, I did check for updates, but it told me that 2.500 is the newest version available.

**Comment by fifafu** [comment]
Possibly check for alpha version updates, the 2.503 version is not yet being pushed actively. In that version the non-background version should also work.

I have just tried to reproduce it, but seem to be unable to get BTT to freeze.

Thes can be copied & pasted into BTT:


The HTML I used (with a three finger tap gesture):
```
{
  "BTTTriggerType" : 104,
  "BTTTriggerTypeDescription" : "3 Finger Tap",
  "BTTTriggerClass" : "BTTTriggerTypeTouchpadAll",
  "BTTPredefinedActionType" : 249,
  "BTTPredefinedActionName" : "Show Floating WebView\/HTML Menu",
  "BTTActionWebViewSize" : "Floating",
  "BTTActionFloatingHTMLConfig" : "{\"BTTCloseOnOutsideClick\":false,\"BTTUseWhiteBackground\":false,\"BTTCloseOnBrowserOpen\":false,\"BTTShowButtons\":true,\"BTTSize\":\"{300, 300}\",\"BTTDoNotCache\":false}",
  "BTTEnabled2" : 1,
  "BTTEnabled" : 1,
  "BTTOrder" : 0,
  "BTTFiles" : [
    {
      "BTTFileContent" : "PGRpdiBpZD0idGVzdCI+PC9kaXY+CjxidXR0b24gb25DbGljaz0id2luZG93LkJUVC5jYWxsSGFuZGxlcigndHJpZ2dlcl9uYW1lZCcsCiAgICAgICAgewogICAgICAgICAgICB0cmlnZ2VyX25hbWU6ICdBY3Rpb24yJywKICAgICAgICAgICAgY2xvc2VGbG9hdGluZ0hUTUxNZW51OiAwCiAgICAgICAgfSwgZnVuY3Rpb24ocmVzcG9uc2UpIHsKICAgICAgICBkb2N1bWVudC5nZXRFbGVtZW50QnlJZCgndGVzdCcpLmlubmVySFRNTCA9IHJlc3BvbnNlOwoKfSkiPgogICAgICAgIFRlc3QgVHJpZ2dlcgogICAgPC9idXR0b24+",
      "BTTFileOther" : "html"
    }
  ]
}
```


The Named Trigger I used:
```
{
  "BTTTriggerType" : 643,
  "BTTTriggerTypeDescription" : "Named Trigger: Action2",
  "BTTTriggerClass" : "BTTTriggerTypeOtherTriggers",
  "BTTPredefinedActionType" : 172,
  "BTTPredefinedActionName" : "Run Apple Script (blocking)",
  "BTTInlineAppleScript" : "return (output volume of (get volume settings))",
  "BTTTriggerName" : "Action2",
  "BTTEnabled2" : 1,
  "BTTEnabled" : 1,
  "BTTOrder" : 0
}
```

**Comment by CalderH** [comment]
All right, I updated to 2.503, and that was all I needed to solved the issue — I was able to make the non-background version work.

Now that it does work, what happens is that I do a two-finger swipe on the Touch Bar to change the volume, and every time the volume changes, the Floating Web View flips between on and off, so that changing the volume makes it flash many times. I was trying to figure out if there was a way to only show the Floating Web View once, and then refresh it every time the volume changes, but I couldn’t see how to do that. It seems that there’s no way to have it constantly checking the result of the AppleScript and updating itself based on that. Am I correct that this is not currently possible?

Thanks for your help!
